official/wasm/cmd/cmd.go

```diff
@@ -0,0 +1,127 @@
+package cmd
+
+import (
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/ignite/cli/v28/ignite/pkg/cliui/colors"
+	"github.com/ignite/cli/v28/ignite/pkg/xgenny"
+	"github.com/ignite/cli/v28/ignite/services/chain"
+	"github.com/spf13/cobra"
+)
+
+// NewWasm creates a new wasm command that holds
+// some other sub commands related to CosmWasm.
+func NewWasm() *cobra.Command {
+	c := &cobra.Command{
+		Use:           "wasm [command]",
+		Short:         "Ignite wasm integration",
+		Aliases:       []string{"w"},
+		SilenceUsage:  true,
+		SilenceErrors: true,
+	}
+
+	// add sub commands.
+	c.AddCommand(
+		NewWasmAdd(),
+		NewWasmConfig(),
+	)
+	return c
+}
+
+const (
+	flagPath = "path"
+	flagHome = "home"
+
+	statusScaffolding  = "Scaffolding..."
+	statusAddingConfig = "Adding config..."
+)
+
+var (
+	modifyPrefix = colors.Modified("modify ")
+	createPrefix = colors.Success("create ")
+	removePrefix = func(s string) string {
+		return strings.TrimPrefix(strings.TrimPrefix(s, modifyPrefix), createPrefix)
+	}
+)
+
+func flagSetPath(cmd *cobra.Command) {
+	cmd.PersistentFlags().StringP(flagPath, "p", ".", "path of the app")
+}
+
+func flagSetHome(cmd *cobra.Command) {
```

Comment:
We can use `ClientAPI` for `path` and `home` instead of manually getting them as flags from user.

Reply:
[Here](https://github.com/ignite/apps/blob/main/examples/chain-info/main.go#L27) we used `api.GetChainInfo(ctx)` to get `ChainInfo` which contains path and home path of the chain and [here](https://github.com/ignite/apps/blob/main/examples/chain-info/cmd/info.go) we use the chain object made from the `AppPath`.

Reply:
@jeronimoalbi If we don't redefine the `--path` flag here in the app would it still inherit it from ignite global flags?

Reply:
It won't the `--path` flag must be defined in the app commands too